aicssegmentation/structure_wrapper/WorkflowStep.py

```diff
@@ -0,0 +1,53 @@
+import importlib
+import numpy as np
+from typing import Dict, List
+
+class WorkflowStep:
+    """
+    A class that defines a step in an AICS-Segmentation workflow.
+    """
+    def __init__(self, step_config: Dict):
+        """
+        Constructor for the workflow object
+
+        Params:
+            step_config (dict): dictionary object containing information about this workflow step
+        """
+        self.name: str = step_config["name"]         # Name of the workflow step
+        self.parent: List = None
+        if isinstance(step_config["parent"], int):
+            # single parent
+            self.parent = [step_config["parent"] - 1] # Index of parent in entire workflow #TODO: Better to change json to 0-indexed to avoid confusion
+        else:
+            # multiple parents
+            self.parent = [i - 1 for i in step_config["parent"]]
+        self.result: np.ndarray = None               # Result of running this step, None if not executed
+
+        module = importlib.import_module(step_config["module"])
+        self.__function = getattr(module, step_config["function"])
+        self.__parameters: dict = None
+        if "parameter" in step_config:
+            self.__parameters = step_config["parameter"]
+
+    def execute(self, image: List) -> np.ndarray:
+        """
+        Execute this workflow step on a given image and return the result.
+        Also sets the result field to the resultant image.
+
+        Params:
+            image (np.ndarray): Image to perform this workflow step on, generally parent image
+
+        Returns:
+            self.result (np.ndarray): Result of performing workflow step on the given image.
+       """
+        if self.__parameters:
+            self.result: np.ndarray = self.__function(*image, **self.__parameters)
+        else:
+            # for "Merge Segmentation' we need a list of images, while other functions take in with multiple params
+            if self.name == "Merge Segmentation":
```

Comment:
I'm not a huge fan of hard-coding some logic based on the name of a step. Can we come up with something more explicit / robust, for example a property on the WorkflowStep to mark whatever is going on here?

Reply:
Hm, I wonder if there is a way to get the function we're calling, and see what it expects as parameters. This was the single exception where we need a list of images so I thought hard coded it for now. Will look into a cleaner solution

Reply:
our current solution on this issue is something like this: https://github.com/AllenCell/aics-segmentation/blob/feature/json_workflow/aicssegmentation/structure_wrapper_config/structure_config_utils.py#L37

basically, just check if the parent is a list or not. 

Also, "Merge Segmentation" is not the only one that takes mulitple inputs. There are quite a few more. For example https://github.com/AllenCell/aics-segmentation/blob/feature/json_workflow/aicssegmentation/structure_wrapper_config/conf_lmnb1_interphase.json#L78

Reply:
My code already supports multiple inputs in function parameters. I think my issue is different- I need to know whether a function expects multiple images as one list(eg. the `segmentation_union()` function) or multiple images unpacked into individual parameters (eg. the `get_seed_for_objects()` function) I tried your code on the lmnb1_interphase workflow and still got this error on `segmentation_union()`
![image](https://user-images.githubusercontent.com/43979182/111234803-9c6a1880-85ac-11eb-83de-eecbb6a3f6b0.png)

Reply:
Would this be a good place to use a try/except? This would work:
```
            try:
                # Most functions require unpacking the images into individual params
                self.result: np.ndarray = self.__function(*image)
            except:
                # Some functions want it as a list
                self.result: np.ndarray = self.__function(image)
```

Reply:
looks ok to me

Reply:
I think that's an ok workaround for now. My question would be: could we instead standardize the functions so that they all either take a List of images, or as multiple params - but not both? If all functions take images as inputs, it should make sense to standardize the input format to stay consistent.